### PR TITLE
Fix namespace collision and parameter_callback problems in PidROS

### PR DESCRIPTION
**Pull request by progtologist**
This fixes problems in PidROS with using the parameters (reconfiguration fails when the node has extra parameters declared) and with namespacing issues (topic namespace argument can lead to problems if defined deeper than one)

### CODE REVIEW

File: src/pid_ros.cpp [hunk 1]
```diff
@@ -294,40 +312,45 @@ PidROS::setParameterEventCallback()
...
+          } else if (param_name == param_prefix_ + "i_clamp_min") {
             gains.i_min_ = parameter.get_value<double>();
-          } else if (param_name == topic_prefix_ + "antiwindup") {
+            changed = true;
+          } else if (param_name == param_prefix_ + "antiwindup") {
             gains.antiwindup_ = parameter.get_value<bool>();
-          } else {
-            result.successful = false;
-            result.reason = "Invalid parameter";
+            changed = true;
```
Comment (destogl) [inline src/pid_ros.cpp:337]: Why this change?

**Review by destogl — APPROVED** [review]
Very nice solution without API breaks! Thanks!

File: src/pid_ros.cpp (hunk 1 above)
Comment (progtologist) [reply src/pid_ros.cpp:337]: The node_params_->add_on_set_parameters_callback() adds a callback that will be called whenever any parameter of the node changes. E.g. if you have 2 instances of PidROS, or if you have your own parameters in the same node, then this callback will be called even if a parameter not declared by PidROS changes. If we set result.successful to false because the parameter is not known to PidROS, we are setting false the reconfiguration of the node in general.
